README.md

```diff
@@ -1,11 +1,155 @@
-# lambda-template
-A GitHub template for quickly starting a new AWS lambda project.
-
-## Naming
-Naming conventions:
-* for a vanilla Lambda: `lambda-<context>`
-* for a Cloudformation Transform macro: `cfn-macro-<context>`
-* for a Cloudformation Custom Resource: `cfn-cr-<context>`
+# lambda-finops-cost-rules
+A lambda for generating the JSON string used by the Rules property of an
+AWS::CE::CostCategory CloudFormation resource.
+
+## Design
+This lambda is intended to transform the chart of accounts returned from
+[lambda-mips-api](https://github.com/Sage-Bionetworks-IT/lambda-mips-api),
+combined with current AWS account tags, into cost-category rules (JSON string)
+appropriate to pass to the [`Rules` property for `AWS::CE::CostCategory`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ce-costcategory.html#cfn-ce-costcategory-rules).
+
+### Category Rules
+This lambda will create cost-category rules that check a list of tags (provided
+as a template parameter) for each account code listed in our chart of accounts,
+also create rules for each account tagged with a listed tag, and finally create
+rules to inherit the value from a listed tag.
+
+The order of the rules matters because processing will stop at the first match.
+For more information on rule processing, see the [AWS User Guide](https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/manage-cost-categories.html).
+
+### Components
+![Component Diagram](docs/component.diagram.png)
+
+### Template Parameters
+The following template parameters are used to configure CloudFormation resources:
+| Template Parameter | Description |
+| --- | --- |
+| AcmCertificateArn | ACM certificate when using custom DNS names with Cloudfront |
+| DnsNames | List of custom DNS names for Cloudfront |
+| CacheTTL | TTL value for CloudFront cache objects. |
+
+The following template parameters are passed through as environment variables
+| Template Parameter | Environment Variable | Description |
+| --- | --- | --- |
+| TagList | TagList | List of tag names that may contain category information |
+
+### Triggering
+The CloudFormation template will output the available endpoint URL for triggering the lambda, e.g.:
+`https://abcxyz.cloudfront.net/rules`
+
+These URLs can be also constructed by appending the API Gateway paths to the CloudFormation domain.
+
+#### Origin URL
+The CloudFormation template also outputs the origin URL behind the CloudFront distribution for debugging purposes.
+
+### Response
+This lambda will produce the example output when provided with the example input and default parameters:
+
+#### Example Input
+Example chart of accounts returned from `lambda-mips-api`:
```

Comment:
this is a bit confusing because it does not provide any info on how to get these values, it might help if you provide a reference to the API to get this data?

Also there's no info on how this `input` is to be used.  Will this get passed into the lambda via the `TagList` ENV VAR?  if so, it would really help if you can provide a snippet of the template showing how that works.  You can do that now or after you've  implement the lambda.

Reply:
Ah, I'm missing a template parameter for the mips api URL, I'll add that and expand on this section a bit to explain that it's querying another lambda

Reply:
so what is the link, if any, between this info and the template parameters?  could you please explain in the docs?